File: LC-Python-Solution/167. Two Sum II - Input Array Is Sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:

        res = []
        l=0
        r=len(numbers)-1

        while l<r:
            if numbers[l] + numbers[r] > target:
                r-=1
            elif numbers[l] + numbers[r] < target:
                l+=1
            elif numbers[l]+numbers[r]==target:
                res.append(l+1)
                res.append(r+1)
                return res
    
        

File: LC_JS-SOLUTIONS/167. Two Sum II - Input Array Is Sorted.js
/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (numbers, target) {
  let l = 0;
  let r = numbers.length - 1;
  let res = [];

  while (l < r) {
    if (numbers[l] + numbers[r] > target) {
      r -= 1;
    } else if (numbers[l] + numbers[r] < target) {
      l += 1;
    } else if (numbers[l] + numbers[r] === target) {
      res.push(l + 1);
      res.push(r + 1);
      return res;
    }
  }
};
